sklearn/impute/_base.py

```diff
@@ -612,7 +612,23 @@ def fit(self, X, y=None):
             raise ValueError("'sparse' has to be a boolean or 'auto'. "
                              "Got {!r} instead.".format(self.sparse))
 
-        self.features_ = self._get_missing_features_info(X)[1]
+        return self._get_missing_features_info(X)
```

Comment:
So that `_fit` actually fits and learns something:

```py
missing_features_info = self._get_missing_features_info(X)
self.features_ = missing_features_info[1]
return missing_features_info
```